src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelRegulationLegend.tsx

```diff
@@ -42,6 +43,22 @@ const TrackPanelRegulationLegend = (props: { disabled: boolean }) => {
   const dispatch = useDispatch();
   const { trackDrawerOpened, reportTrackPanelSectionToggled } =
     useGenomeBrowserAnalytics();
+  const { data: regulationStats } = useGetSpeciesStatisticsQuery(
+    {
+      genomeId: activeGenomeId ?? ''
+    },
+    {
+      skip: !activeGenomeId
+    }
+  );
+
+  if (!regulationStats) {
+    return;
+  }
+  const availableRegulatoryFeatures = regulationLegend.filter((legend) => {
+    return Object.keys(regulationStats).includes(legend.id);
```

Comment:
No need to repeatedly create arrays with the same values (`Object.keys`) inside of the loop. Create the array once outside of the loop.